include/integration/simple_MC_integration.hpp

```diff
@@ -0,0 +1,283 @@
+// VolEsti (volume computation and sampling library)
+
+// Copyright (c) 2012-2021 Vissarion Fisikopoulos
+// Copyright (c) 2018-2021 Apostolos Chalkis
+
+// Contributed and/or modified by Suraj Choubey, as part of Google Summer of Code 2021 program.
+
+// Licensed under GNU LGPL.3, see LICENCE file
+
+// Monte Carlo Integration algorithm used here : https://en.wikipedia.org/wiki/Monte_Carlo_integration#Overview
+
+#ifndef SIMPLE_MC_INTEGRATION_HPP
+#define SIMPLE_MC_INTEGRATION_HPP
+
+#include <iostream>
+#include <fstream>
+#include <random>
+#include <vector>
+#include "convex_bodies/hpolytope.h"
+#include "Eigen/Eigen"
+#include "generators/known_polytope_generators.h"
+#include "boost_random_number_generator.hpp"
+#include "cartesian_geom/cartesian_kernel.h"
+#include "random_walks/random_walks.hpp"
+#include "volume/volume_sequence_of_balls.hpp"
+#include "volume/volume_cooling_gaussians.hpp"
+#include "volume/volume_cooling_balls.hpp"
+#include "misc.h"
+
+typedef double NT;
+typedef Cartesian<NT> Kernel;
+typedef typename Kernel::Point Point;
+typedef std::vector<Point> Points;
+typedef HPolytope<Point> HPOLYTOPE;
+typedef boost::mt19937 RNGType;
+typedef BoostRandomNumberGenerator<RNGType, NT> RandomNumberGenerator;
+typedef typename HPolytope<Point>::MT MT;
+typedef typename HPolytope<Point>::VT VT; 
+
+typedef const unsigned int Uint;  // Positive constant value for no of samples & dimensions
+enum volumetype {CB ,CG ,SOB}; // Volume type for polytope
+typedef typename std::vector<NT> Limit; // Standard way for user to use limits 
+// E.g. Limits LL{0.5, 1.5, 2.5} ; Limits UL{1.2, 1.8 , 2.8 } ;
+
+const Limit lt{0}; // To initialize non-initialized limits
+const Point pt{0}; // To initialize non-initialized points
+
+// To check if two n-dimensional points ensure valid limits in integration
+template
+<
+    typename Point = Point,
+    typename NT = NT
+>
+bool valid_limits(Point LL, Point UL) {
+    if (UL.dimension() == LL.dimension()) {
+        for (int i = 0; i<LL.dimension(); i++) {
+            if (LL[i] > UL[i]) {
+                std::cerr << "Invalid integration limits\n";
+                return false;
+            }
+        }
+        return true;
+    } else {
+        std::cerr << "Invalid integration limits\n";
+        return false;
+    }
+}
+
+// Initialize Limit Point
+template
+<
+    typename Point = Point,
+    typename NT = NT
+>
+Point init_limit(Limit L, Uint dim) {
+    Point pt(dim);
+    for (int i=0; i<dim; i++) {
```

Comment:
comparison of integers of different signs: 'int' and 'Uint' (aka 'const unsigned int'), I think it is ok to use `int` everywhere